### DISCUSSION TITLE
RKE2.yaml permissions

### DISCUSSION THREAD

**Discussion by dpearceMN** [rancher/rke2#2773] [Q&A]
Switching from plain K8S to rke2 version v1.22.8+rke2r1 (bd020f4f0c54f8dcd99fc26a7ea503367741c99d) go version go1.16.14b7.

Perhaps I am missing something real basic, but am I supposed to be able to run "kubectl get nodes" **without** being root?

The file permissions on /etc/rancher/rke2/rke2.yaml keeps getting reset to 0600, but then kubectl  can't read the file. If I change the permissions to 0644, the command works without being root.

Thanks!

**Comment by dpearceMN** [comment]
So if I run "sudo kubectl get nodes **--kubeconfig=/etc/rancher/rke2/rke2.yaml**" It will work.

The endowment variable KUBECONFIG was not being set properly.

**So in RKE2, you have to run kubectl as root?**

**Reply by brandond** [reply]
Unless you change the [`write-kubeconfig-mode` ](https://docs.rke2.io/install/install_options/server_config/#:~:text=write%2Dkubeconfig%2Dmode) option, yes. Having your cluster admin kubeconfig be world-readable by default is not generally something that people want.

A better idea might be to copy it to your non-root users ~/.kube/ directory.
